SYCLStream.cpp

```diff
@@ -47,7 +47,15 @@ SYCLStream<T>::SYCLStream(const unsigned int ARRAY_SIZE, const int device_index)
   std::cout << "Driver: " << getDeviceDriver(device_index) << std::endl;
   std::cout << "Reduction kernel config: " << dot_num_groups << " groups of size " << dot_wgsize << std::endl;
 
-  queue = new cl::sycl::queue(dev);
+  queue = new cl::sycl::queue(dev, [&](cl::sycl::exception_list l) {
+      try {
+        for(auto e: l) {
+          std::rethrow_exception(e);
+        }
+      } catch (cl::sycl::exception e) {
+        std::cout << e.what();
```

Comment:
This is a fair point. The code now prints all SYCL exceptions and then throws an `std::runtime_exception`. b8f7a5427e82068f61b43373a8c59c6111858b35